code/streaming/Hacl.Streaming.SHA2.fst

```diff
@@ -62,42 +62,56 @@ let hash_t (a: sha2_alg) =
     (ensures  fun h0 _ h1 -> modifies (loc dst) h0 h1 /\
       as_seq h1 dst == Spec.Agile.Hash.hash a (as_seq h0 input))
 
-inline_for_extraction noextract
-let alloca_224 = F.alloca hacl_sha2_224 () (state_t_224.s ()) (G.erased unit)
-let create_in_224 = F.create_in hacl_sha2_224 () (state_t_224.s ()) (G.erased unit)
-let init_224 = F.init hacl_sha2_224 (G.hide ()) (state_t_224.s ()) (G.erased unit)
-[@@ Comment "0 = success, 1 = max length exceeded" ]
-let update_224 = F.update hacl_sha2_224 (G.hide ()) (state_t_224.s ()) (G.erased unit)
-let finish_224 = F.mk_finish hacl_sha2_224 () (state_t_224.s ()) (G.erased unit)
-let free_224 = F.free hacl_sha2_224 (G.hide ()) (state_t_224.s ()) (G.erased unit)
-
 open Lib.NTuple
 open Lib.MultiBuffer
 open Hacl.Spec.SHA2.Vec
 open Hacl.SHA2.Scalar32
 open Hacl.Impl.SHA2.Generic
 module ST = FStar.HyperStack.ST
 
-val sha224: hash_t SHA2_224
-let sha224 dst input_len input =
-  let ib = ntup1 input in
-  let rb = ntup1 dst in
-  let h0 = ST.get() in
-  loc_multi1 rb;
-  hash #SHA2_224 #M32 sha224_init sha224_update_nblocks sha224_update_last sha224_finish rb input_len ib;
-  let h1 = ST.get() in
-  Hacl.Spec.SHA2.Equiv.hash_agile_lemma #SHA2_224 #M32 (v input_len) (as_seq_multi h0 ib);
-  assert ((as_seq_multi h1 rb).(|0|) == as_seq h1 dst)
+// SHA2-256
+// --------
 
 inline_for_extraction noextract
 let alloca_256 = F.alloca hacl_sha2_256 () (state_t_256.s ()) (G.erased unit)
+
+[@@ Comment
+"Allocate initial state for the SHA2_256 hash. The state is to be freed by
+calling `free_256`."]
 let create_in_256 = F.create_in hacl_sha2_256 () (state_t_256.s ()) (G.erased unit)
+
+[@@ Comment
+"Reset an existing state to the initial hash state with empty data."]
 let init_256 = F.init hacl_sha2_256 (G.hide ()) (state_t_256.s ()) (G.erased unit)
-[@@ Comment "0 = success, 1 = max length exceeded" ]
+
+[@@ Comment
+"Feed an arbitrary amount of data into the hash. This function returns 0 for
+success, or 1 if the combined length of all of the data passed to `update_256`
+(since the last call to `init_256`) exceeds 2^63-1 bytes.
+
+This function is identical to the update function for SHA2_224.";
+
+CPrologue
+"#define Hacl_Streaming_SHA2_update_224 Hacl_Streaming_SHA2_update_256" ]
```

Comment:
Relying on an untrusted define for this is a bit annoying. This seems fine for SHA2, as the specs do not change, and we can reasonably assume that update_224 will always have exactly the same signature as update_256, but it might be better to check for it in krml instead, and automatically generate the #define based on the F* code.
Beyond this PR, what would you think for instance of a CDefine/CAbbrvDefine (name to be determined) feature in krml doing exactly that?